addons/point_of_sale/models/pos_config.py

```diff
@@ -0,0 +1,239 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import uuid
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class PosConfig(models.Model):
+    _name = 'pos.config'
+
+    POS_CONFIG_STATE = [
+        ('active', 'Active'),
+        ('inactive', 'Inactive'),
+        ('deprecated', 'Deprecated')
+    ]
+
+    def _default_sale_journal(self):
+        return self.env['account.journal'].search([('type', '=', 'sale'), ('company_id', '=', self.env.user.company_id.id)], limit=1)
+
+    def _default_pricelist(self):
+        return self.env['product.pricelist'].search([], limit=1)
+
+    def _get_default_location(self):
+        return self.env['stock.warehouse'].search([('company_id', '=', self.env.user.company_id.id)], limit=1).lot_stock_id
+
+    def _get_default_company(self):
+        return self.env['res.users']._get_company()
+
+    def _get_default_nomenclature(self):
+        return self.env['barcode.nomenclature'].search([], limit=1)
```

Comment:
You don't think, its gives always unpredictable result, If i am wrong then please correct me.